tools/python/format_checker.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import print_function
+
+import logging
+import subprocess
+
+from integration_tests_runner import setup_jenkins_console_logger
+
+
+FILE_EXTENSIONS = [".cpp", ".hpp"]
+EXCLUDE_FOLDERS = ["base"]
+
+setup_jenkins_console_logger()
+
+def git(command):
+    return exec_shell("git", command).strip()
+
+def clang_format(source_file):
+    return exec_shell("clang-format", "-i {0}".format(source_file))
+
+def exec_shell(command, flags):
+    spell = "{0} {1}".format(command, flags)
+
+    process = subprocess.Popen(
+        [spell],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        shell=True
+    )
+
+    logging.getLogger().info(spell)
+    out, _ = process.communicate()
+    return out
+
+
+def current_commit():
+    return git("rev-parse HEAD")
+
+
+def main_parent_of(sha):
+    return git("show --format='%H' {0}^1".format(sha))
+
+
+def files_in_diff():
+    cur_commit = current_commit()
+    main_parent = main_parent_of(cur_commit)
+    diff_files = git(
+        "diff --name-only {cur} {parent}".format(
+            cur=cur_commit, parent=main_parent
+        )
+    )
+
+    return filter(filter_files, diff_files)
+
+
+def filter_files(file_name):
+    filtered_extensions = filter(lambda s: file_name.endswith(s), FILE_EXTENSIONS)
+    if not filtered_extensions:
+        return False
+
+    filtered_folders = filter(lambda s: file_name.startswith(s), EXCLUDE_FOLDERS)
```

Comment:
Please, separate file_name to parts and check that the first part is a directory and not in exclusion list.